pkg/accumulator/accumulator.go

```diff
@@ -0,0 +1,221 @@
+	for i := range acc.accumulatorData {
+		if len(acc.accumulatorData[i]) == 0 {
+			acc.createNewMap(event, i)
+			return i
+		}
+		firstKey := acc.firstMapKeysOfAccumulatorData[i]
+		if event.GetEventTimestamp().Sub((acc.accumulatorData[i])[firstKey][0].GetEventTimestamp()) < time.Second {
+			return i
+		}
+	}
+	index := acc.findIndexByTimestampWhenAccumulatorDataIsFull(event)
+	if index != -1 {
+		return index
+	}
+	// logger.L().Debug("findIndexByTimestamp: failed to find index, sniffer data will not saved")
+	return -1
+}
+
+func (acc *CacheAccumulator) removeAllStreamedContainers(event *evData.EventData) {
+	acc.containersData.registerMutex.Lock()
+	if len(acc.containersData.accumulatorDataPerContainer) > 0 {
+		for contID := range acc.containersData.accumulatorDataPerContainer {
+			acc.containersData.accumulatorDataPerContainer[contID] <- *event
+		}
+	}
+	acc.containersData.registerMutex.Unlock()
+}
+
+func (acc *CacheAccumulator) addEventToCacheAccumulator(event *evData.EventData, index int) {
+	acc.syncReaderWriterAccumulatorData.Lock()
+	a := acc.accumulatorData[index]
+	a[event.GetEventContainerID()] = append(a[event.GetEventContainerID()], *event)
+	acc.accumulatorData[index][event.GetEventContainerID()] = append(acc.accumulatorData[index][event.GetEventContainerID()], *event)
+	acc.syncReaderWriterAccumulatorData.Unlock()
+}
+
+func (acc *CacheAccumulator) streamEventToRegisterContainer(event *evData.EventData, index int) {
+	acc.containersData.registerMutex.Lock()
+	if containerAccumulatorChan, exist := acc.containersData.accumulatorDataPerContainer[event.GetEventContainerID()]; exist {
+		containerAccumulatorChan <- *event
+	}
+	acc.containersData.registerMutex.Unlock()
+}
+
+func (acc *CacheAccumulator) accumulateEbpfEngineData() {
+	for {
+		event := <-acc.mainDataChannel
+		logger.L().Debug("metadataAcc ", helpers.String("", fmt.Sprintf("%v", event)))
+		if strings.Contains(event.GetEventContainerID(), myContainerID) {
+			continue
+		}
+		if event != nil {
+			if event.GetEventCMD() == "drop event occurred\n" {
+				acc.removeAllStreamedContainers(event)
+			} else {
+				index := acc.findIndexByTimestamp(event)
+				if index == -1 {
+					// logger.L().Debug("metadataAcc %v\n", metadataAcc)
+					continue
+				}
+				acc.addEventToCacheAccumulator(event, index)
+				acc.streamEventToRegisterContainer(event, index)
+			}
+		}
+	}
+}
+
+func (acc *CacheAccumulator) getEbpfEngineData() {
+	acc.ebpfEngine.GetData(acc.mainDataChannel)
+}
+
+func (acc *CacheAccumulator) getEbpfEngineError(errChan chan error) {
+	errChan <- acc.ebpfEngine.GetEbpfEngineError()
+}
+
+func (acc *CacheAccumulator) StartAccumulator(errChan chan error) error {
+	falcoEbpfEngine := ebpfeng.CreateFalcoEbpfEngine(config.GetConfigurationConfigContext().GetSyscallFilter(), false, false, "")
+	acc.ebpfEngine = falcoEbpfEngine
+
+	err := acc.ebpfEngine.StartEbpfEngine()
+	if err != nil {
+		logger.L().Error("fail to create ebpf engine")
```

Comment:
if you are logging the error here, you should also include the original error in the log